### PR TITLE
code to ingest real-time weather data as CSVs in S3

### PR DESCRIPTION
**Pull request by Retr0Metal98**
Wrote a script to run through the data.gov.sg weather APIs and ingest the real-time data (will update later to include functions to collect weather forecasts and pollution data)

Will need to refactor data format as needed by model team, and for ingestion into AWS S3 and for use by the spark system to accept.

### CODE REVIEW

**Review by don-tay — COMMENTED** [review]
For the file `lta_camera.py`, it is essentially what `data_ingest.py` is already doing. The main difference is that lta_camera.py stores the file locally while `data_ingest.py` uploads this same file to S3.

Would you like the local file storage functionality to be implemented as well?

File: weather_data.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        print("Collected info into dataframe!")
+
+    return real_time_dfs
+
+
+# try the above function here
+real_time_weather_dfs = real_time_weather()
+for (field, df) in real_time_weather_dfs.items():
+    print("Field:", field)
+    print(df)
```
Comment (don-tay) [inline weather_data.py]: Should we extend this script to store the results into a relational db?

**Comment by Retr0Metal98** [comment]
I guess we can keep the local file storage functionality to use for quick testing purposes? Perhaps we can merge the two files and let the user toggle between AWS storage/local storage somehow?

Btw, I think don't merge the branches yet, I can add more functions for the other data sources (weather forecast, pollution data, etc.)

**Comment by don-tay** [comment]
> I guess we can keep the local file storage functionality to use for quick testing purposes? Perhaps we can merge the two files and let the user toggle between AWS storage/local storage somehow?
> 
> Btw, I think don't merge the branches yet, I can add more functions for the other data sources (weather forecast, pollution data, etc.)

Yes this is possible, we can use an env flag to toggle between `local` and `development`. That said, I suggest we see what works best for the development flow in the model team

File: helper.py [hunk 2]
```diff
@@ -32,9 +45,91 @@ def get_req_handler(url: str) -> requests.Response:
...
+        if s3_bucket:
+            try:
+                # upload dataframe to S3 bucket
+                csv_buffer = StringIO()
+                dataframe.to_csv(csv_buffer, encoding="utf-8")
+                s3_bucket.put_object(Body=csv_buffer.getvalue(), Key=savepath)
+            except Exception:
+                print("Failed to upload data in " + savepath, file=sys.stderr)
+            else:
+                print("Uploaded dataframe as csv file " + savepath + " to bucket!")
```
Comment (don-tay) [inline helper.py]: Shortening this portion with the same effect
```suggestion
                s3_bucket.put_object(Body=csv_buffer.getvalue(), Key=savepath)
                print("Uploaded dataframe as csv file " + savepath + " to bucket!")
            except Exception:
                print("Failed to upload data in " + savepath, file=sys.stderr)
```

File: helper.py [hunk 3]
```diff
@@ -32,9 +45,91 @@ def get_req_handler(url: str) -> requests.Response:
...
+        else:
+            print("When using output='AWS', specify S3.Bucket object in s3_bucket.")
```
Comment (don-tay) [inline helper.py]: Cutting down on nested conditions with guard clause
```suggestion
        if not s3_bucket:
            print("When using output='AWS', specify S3.Bucket object in s3_bucket.")
            return
            
              try:
                  # upload dataframe to S3 bucket
                  csv_buffer = StringIO()
                  dataframe.to_csv(csv_buffer, encoding="utf-8")
                 s3_bucket.put_object(Body=csv_buffer.getvalue(), Key=savepath)
             except Exception:
                 print("Failed to upload data in " + savepath, file=sys.stderr)
```

File: lta_camera.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
+import os
```
Comment (don-tay) [inline lta_camera.py]: We can delete this file since `data_ingest.py` can be extended to include local file upload

File: lta_camera.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    if img_request.ok:
+        img_data = img_request.content
+        img_dir = os.path.join(os.curdir, save_dir, cam_id)
+        if not os.path.exists(img_dir):
+            os.makedirs(img_dir)  # create subdirectory for each camera id
+        img_file_path = os.path.join(img_dir, timestamp + ".jpg")
+
+        with open(img_file_path, 'xb') as handler:
+            handler.write(img_data)
+        handler.close()
```
Comment (don-tay) [inline lta_camera.py]: When `with open(...)...` is used, the file `close()` will be handled automatically. Hence there is no need to call the `close()` method
```suggestion
        with open(img_file_path, 'xb') as handler:
            handler.write(img_data)
```

**Review by don-tay — APPROVED** [review]
LGTM, some code refactor, and i think we can do away with `lta_camera.py` and we are good to go

**Pull request merged**